File: Programmers/LEVEL 1/완주하지못한선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string, int> umap;
    unordered_map<string, int>::iterator it;
    
    for (int i = 0; i < participant.size(); i++) umap[participant[i]]++;
    for (int i = 0; i < completion.size(); i++) umap[completion[i]]--;
    for (it = umap.begin(); it != umap.end(); it++) {
        if (it->second != 0) answer = it->first;
    }
    
    return answer;
}

File: Programmers/코딩테스트 고득점 Kit/Hash/완주하지못한선수.java
import java.util.Set;
import java.util.HashMap;
import java.util.Iterator;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> hm = new HashMap<>();

        for (String member : participant)
            hm.put(member, hm.getOrDefault(member, 0) + 1);
        for (String member : completion)
            hm.put(member, hm.get(member) - 1);
        Set<String> keys = hm.keySet();
        Iterator<String> it = keys.iterator();

        while (it.hasNext()) {
            String key = it.next();
            int count = hm.get(key);

            if (count > 0)
                answer += key;
        }

        return answer;
    }
}